src/frontend/src/components/common/Table/index.tsx

```diff
@@ -102,23 +104,34 @@ export const Table = ({
                         key={column.key + '-' + index}
                         className={column.className || ''}
                       >
-                        <InputText
-                          type="text"
-                          value={rowData?.[column.key]}
-                          labelSide="left"
-                          label={
-                            column.numberedLabelPrefix &&
-                            `${column.numberedLabelPrefix} ${index + 1}`
-                          }
-                          isDisabled={column.isDisable}
-                          onChange={(e) =>
-                            handleDataChange(e, `${index}.${column.key}`)
-                          }
-                        />
+                        {!readOnly ? (
+                          <InputText
+                            type="text"
+                            value={rowData?.[column.key]}
+                            labelSide="left"
+                            label={
+                              column.numberedLabelPrefix &&
+                              `${column.numberedLabelPrefix} ${index + 1}`
+                            }
+                            isDisabled={column.isDisable}
+                            onChange={(e) =>
+                              handleDataChange(e, `${index}.${column.key}`)
+                            }
+                          />
+                        ) : (
+                          <p>
+                            {column.numberedLabelPrefix && (
+                              <label>
+                                {column.numberedLabelPrefix} {index + 1}
+                              </label>
+                            )}
+                            {rowData?.[column.key]}
+                          </p>
+                        )}
                       </td>
                     ))}
 
-                    {allowRowDelete && (
+                    {allowRowDelete && !readOnly && (
```

Comment:
instead of having multiple if/else, can the 'if (!readonly)'  all be chucked together and the 'else' (readonly \<p\> be declared once?